### PR TITLE
feat: integrate openedx search api with existing CoursewareSearch.jsx

### PR DESCRIPTION
**Pull request by qasimgulzar**
Under this PR I have used [openedx-search-api](https://github.com/qasimgulzar/openedx-search-api) to enable courseware search feature. 

https://github.com/user-attachments/assets/dd591d3f-0091-425c-98af-4634cabe4efa

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @qasimgulzar!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.




#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/committers-frontend-app-learning`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Comment by bradenmacdonald** [comment]
@qasimgulzar Please provide full, detailed instructions for how to test this, including links to all other required PRs and configuration settings.

Personally, I would also really like to see what it looks like when the studio search (in course-authoring) is converted to use this, as it's a far more complex use case than the very simple course search UX here.

**Comment by qasimgulzar** [comment]
Added location information to index
image

**Comment by itsjeyd** [comment]
OSPR management note: Product review in progress via https://github.com/openedx/platform-roadmap/issues/378.

File: webpack.dev.config.js [hunk 1]
```diff
@@ -7,5 +7,6 @@ config.resolve.alias = {
   ...config.resolve.alias,
   '@src': path.resolve(__dirname, 'src'),
 };
+config.externalsType = 'script';
```
Comment (bradenmacdonald) [inline webpack.dev.config.js:10]: What is this for?

**Review by bradenmacdonald — COMMENTED** [review]
For new code like this, I would strongly prefer we load the search engine config and results [using React Query like this](https://github.com/openedx/frontend-app-authoring/blob/4d0f92e265c348c4cdef022ed6986fdb630e55fb/src/search-manager/data/apiHooks.ts) rather than using Redux. See [OEP-67](https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0067/decisions/frontend/0010-react-query.html).

**Comment by regisb** [comment]
Note to self: to test this PR, we have to enable the `courseware.mfe_courseware_search` waffle flag.

**Comment by itsjeyd** [comment]
OSPR management note: Marking this as blocked based on [this update](https://github.com/openedx/platform-roadmap/issues/378#issuecomment-2372089296) from @ormsbee over on the product proposal ticket.

**Comment by itsjeyd** [comment]
Hey @qasimgulzar, could you please check if this PR is still needed now that @regisb created another PR (https://github.com/openedx/edx-platform/pull/35650) for addressing https://github.com/openedx/platform-roadmap/issues/378?

**Comment by regisb** [comment]
Let's close this PR for the moment. The interface that it proposes is different from my PR https://github.com/openedx/edx-platform/pull/35650 because it connects to Meilisearch directly, but we can always re-open it in the future if we decide that we want to go with this approach.

**Comment by itsjeyd** [comment]
Thanks @regisb!

**Pull request closed without merging**